File: solutions/_83_remove_duplicates_from_sorted_list/solution.js
// Definition for singly-linked list.
function ListNode(val, next) {
  this.val = val === undefined ? 0 : val;
  this.next = next === undefined ? null : next;
}

/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function (head) {
  let curr = head;
  while (curr) {
    while (curr.next && curr.next.val == curr.val) {
      curr.next = curr.next.next;
    }
    curr = curr.next;
  }
  return head;
};

module.exports = { deleteDuplicates, ListNode };

/*
https://leetcode.com/problems/remove-duplicates-from-sorted-list/
*/

File: solutions/_83_remove_duplicates_from_sorted_list/Solution.java

package _83_remove_duplicates_from_sorted_list;

class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode curr = head;
        while (curr != null) {
            while (curr.next != null && curr.next.val == curr.val) {
                curr.next = curr.next.next;
            }
            curr = curr.next;
        }
        return head;
    }
}

/*
 * ht /leetcode.com/problems/remove-duplicates-from-sorted-list/
 */
